libraries/AP_Math/control.cpp

```diff
@@ -164,14 +163,22 @@ void shape_vel_accel(float vel_input, float accel_input,
         return;
     }
 
-    // Calculate time constants and limits to ensure stable operation
-    const float KPa = jerk_max / accel_max;
-
     // velocity error to be corrected
     float vel_error = vel_input - vel;
 
+    // Calculate time constants and limits to ensure stable operation
+    float KPa;
+    // The direction of acceleration limit is the same as the velocity error.
+    // This is because the velocity error is negative when slowing down while
+    // closing a positive position error.
+    if (is_positive(vel_error)) {
+        KPa = jerk_max / accel_max;
```

Comment:
No, we only check that it is not zero:
    if (!(is_negative(accel_min) && is_positive(accel_max))) {
        INTERNAL_ERROR(AP_InternalError::error_t::invalid_arg_or_result);
        return;
    }

Reply:
In general we don't check that parameters have been defined to silly low numbers. This is because a silly low number for a small aircraft can be a silly large number for a large aircraft.